src/main/kotlin/org/rust/lang/core/psi/ext/RsPat.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import org.rust.lang.core.psi.*
+
+val RsPat.isIrrefutable: Boolean
+    get() = when (this) {
+        is RsPatTupleStruct -> patList.all { it.isIrrefutable }
+        is RsPatSlice -> patList.all { it.isIrrefutable }
+        is RsPatTup -> patList.all { it.isIrrefutable }
+        is RsPatBox -> pat.isIrrefutable
+        is RsPatRef -> pat.isIrrefutable
+        is RsPatStruct -> patFieldList.all { it.pat?.isIrrefutable ?: it.patBinding != null }
+        is RsPatConst -> false
```

Comment:
Maybe we should return `false` for `RsPatRange` too? (just like in [#3157](https://github.com/intellij-rust/intellij-rust/pull/3157/files#diff-5576698e67fd38b5bfa38c11bd672316R16))

Reply:
Fixed.